pkg/systemd/host.js

```diff
@@ -1338,7 +1359,11 @@ PageSystemInformationChangeSystime.prototype = {
         }
 
         function set_ntp(val) {
-            return self.server_time.timedate.call('SetNTP', [val, true]);
+            var dfd = cockpit.defer();
+            self.server_time.ntp_waiting_promise = dfd;
+            self.server_time.ntp_waiting_value = val;
+            self.server_time.timedate.call('SetNTP', [val, true]);
+            return dfd.promise;
```

Comment:
Let's please avoid introducing new `cockpit.defer()`. Standard JS Promise should be compatible enough.

This should somehow handle failures of `.call()`, at least with a `console.error()`.

Reply:
I need to use defer as I want to resolve this outside of the constructor. So I need to use `Promise.defer()` which is obsoleted. I hoped at least `cockpit.defer()` could be better solution as we can support it on our own.

Reply:
FYI, with standard promises you'd just save the `resolve` handler, like this:
```js
return new Promise((resolve, reject) => { ... self.server_time.ntp_waiting_resolve = resolve; });
```
But `cockpit.defer()` isn't really bad either, of course. Your call.